File: escala_game/lib/models/movement.dart
class Movement {
  final String id;
  final String gameId;
  final String playerId;
  final String actionType;
  final Map<String, dynamic> data;

  Movement({
    required this.id,
    required this.gameId,
    required this.playerId,
    required this.actionType,
    required this.data,
  });

  factory Movement.fromJson(Map<String, dynamic> json) {
    return Movement(
      id: json['_id'],
      gameId: json['gameId'],
      playerId: json['playerId'],
      actionType: json['actionType'],
      data: json['data'] ?? {},
    );
  }
}

File: game_Api/models/Movement.js
const mongoose = require("mongoose");

const MovementSchema = new mongoose.Schema(
  {
    gameId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Game",
      required: true,
    },
    playerId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Player",
      required: true,
    },
    actionType: {
      type: String,
      enum: [
        "JOIN_GAME",
        "PLACE_MATERIAL",
        "MAKE_GUESS",
        "END_TURN",
        "LEAVE_GAME",
      ],
      required: true,
    },
    data: {
      // For PLACE_MATERIAL
      material: {
        type: {
          type: String,
          enum: ["red", "yellow", "green", "blue", "purple"],
        },
        id: String, // opcional, si usas identificadores únicos
      },
      balanceType: { type: String, enum: ["main", "secondary"] },
      side: { type: String, enum: ["left", "right"] },

      // For MAKE_GUESS
      guesses: [
        {
          type: {
            type: String,
            enum: ["red", "yellow", "green", "blue", "purple"],
          },
          weight: Number,
        },
      ],
      guessResult: Boolean,
    },
    clientInfo: {
      ip: String,
      userAgent: String,
    },
  },
  {
    timestamps: true,
  }
);

module.exports = mongoose.model("Movement", MovementSchema);
